core-tests/src/test/java/org/mule/runtime/core/api/type/catalog/ApplicationTypeLoaderTestCase.java

```diff
@@ -24,11 +40,29 @@
 @Story(TYPES_CATALOG)
 public class ApplicationTypeLoaderTestCase extends AbstractMuleTestCase {
 
+  private static final String MOCK_EXTENSION_PREFIX = "mock";
+  private static final String MOCK_TYPE_ALIAS = "MockType";
+  private static final String FULL_CLASS_NAME_FOR_MOCK_TYPE = "full.class.name.for.MockType";
+  private static final String MOCK_EXTENSION_NAME = "Mock Extension";
+
   private ApplicationTypeLoader applicationTypeLoader;
 
   @Before
   public void setUp() {
-    applicationTypeLoader = new ApplicationTypeLoader();
+    ExtensionModel mockExtensionModel = mock(ExtensionModel.class);
+
+    XmlDslModel dslModel = XmlDslModel.builder().setPrefix(MOCK_EXTENSION_PREFIX).build();
+    when(mockExtensionModel.getXmlDslModel()).thenReturn(dslModel);
+
+    when(mockExtensionModel.getName()).thenReturn(MOCK_EXTENSION_NAME);
+
+    Map<Class<? extends TypeAnnotation>, TypeAnnotation> annotationMap = new HashMap<>(2);
+    annotationMap.put(TypeAliasAnnotation.class, new TypeAliasAnnotation(MOCK_TYPE_ALIAS));
+    annotationMap.put(TypeIdAnnotation.class, new TypeIdAnnotation(FULL_CLASS_NAME_FOR_MOCK_TYPE));
+    ObjectType mockType = new DefaultObjectType(emptyList(), false, null, JAVA, annotationMap);
```

Comment:
I believe there are builders for these things. Shouldn't we use them instead?

Should look like this I believe:
```suggestion
    BaseTypeBuilder.create(JAVA).objectType()
      .id(FULL_CLASS_NAME_FOR_MOCK_TYPE)
      .with(new TypeAliasAnnotation(MOCK_TYPE_ALIAS))
      .build()
```

Reply:
Done